climada/hazard/base.py

```diff
@@ -1393,3 +1442,167 @@ def _read_att_excel(self, file_name, var_names):
         self.intensity = sparse.csr_matrix(dfr.values[:, 1:num_events + 1].transpose())
         self.fraction = sparse.csr_matrix(np.ones(self.intensity.shape,
                                                   dtype=np.float))
+
+    def concatenate(self, haz_src, append=False, centroids=None):
+        """
+        Concatenate events of several hazards of same type.
+
+        This is a wrapper method for hazard.base.concatenate_hazard
+        to ensure backwards compatibility.
+
+        Parameters
+        ----------
+        haz_src: list(climada.hazard.Hazard())
+            Hazard instances of the same hazard type
+        append : bool, optional
+            If True, append the haz_src to current object. If False,
+            replace all attributes from current objects by concatenated
+            hazards from haz_src. The default is False.
+        centtroids: climada.hazard.Centroids(), optional
+            Centroids instance on which to map the concatenated hazard.
+            Default is None.
+
+        Returns
+        -------
+        None.
+
+        """
+        if append:
+            haz_src = [self] + haz_src
+        self.__dict__ = copy.deepcopy(
+            concatenate_hazard(haz_src, centroids).__dict__
+            )
+
+
+def join_centroids(haz_list):
+    """
+    Find centroids comon to all.
+
+    Centroids of all hazards must either all be rasters with the same
+    resolution, or all be points.
+
+    The centroids of all hazards are combined together. If points,
+    all points are combined. If rasters, a global raster is defined.
+
+    Parameters
+    ----------
+    haz_list : list(climada.hazard.Hazard())
+        List of hazard with centroids to join
+
+    Returns
+    -------
+    centroids : climada.hazard.Centroids()
+        Centroids containing the centroids of all hazards.
+
+    """
+    centroids = copy.deepcopy(haz_list[0].centroids)
+    for haz in haz_list[1:]:
+        if not centroids.equal(haz.centroids):
+            centroids.append(haz.centroids)
```

Comment:
The duplicates removal on line 1502 might get expensive with large sets of large centroids (I didn't check), so it might also be worth running pairwise checks on the input centroid objects to spot any identical ones.

Reply:
I agree this could be the case, but how expensive is the pair checking between all pairs? The number of pairs growths quickly. So, while I see your point, I am not convinced this is necessary. Do you want me to do some speed tests?

I here test one edge case when all centroids basically are the same (i.e. when someone is merging hazard defined on the same centroids).

Reply:
I think the `centroids.equal` method is very cheap to run. It checks for centroids being the same size first (almost all differing centroids will fail this), and then checks if the centroid `lat` and `lon` arrays are (near-)identical. So unless you're combining hundreds of hazard objects, it should be fast.

Having said that, maybe it doesn't matter - I assume finding the duplicate centroids isn't the slowest part here. I'm happy to leave this alone and wait to see if someone needs to speed it up later.

Reply:
Then I would keep it this way. Thanks!